src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -1194,12 +1194,17 @@ export class ConnectionWidget extends lifecycle.Disposable {
 		let validInputs = this.validateInputs();
 		if (validInputs) {
 			if (this.useConnectionString) {
-				const connInfo = await this._connectionManagementService.buildConnectionInfo(this.connectionString, this._providerName);
-				if (connInfo) {
+				try {
+					const connInfo = await this._connectionManagementService.buildConnectionInfo(this.connectionString, this._providerName);
+					if (!connInfo) {
+						throw Error(this._providerName + ".buildConnectionInfo had an undefined result.");
```

Comment:
That's why it's passed as [`messageDetails`](https://github.com/microsoft/azuredatastudio/blob/main/src/sql/platform/errorMessage/common/errorMessageService.ts#L19), so it not shown but can be rather copied with this button:

<img width="569" alt="grafik" src="https://user-images.githubusercontent.com/951587/226675770-cf911c12-422d-4138-8a6e-d566115955f2.png">

So when running into issues users can provide that error message to support or whoever.

Reply:
1. The general guideline is: anything in the product (ADS + extensions) that is user facing should be localized. external libraries that we don't have control over is ok.
2. if you need something for debugging purpose, you can log it. for example: the `the buildConnectionInfo method returned undefined value.` should go to the log.

@Charles-Gagnon thoughts?

Reply:
Changed with 43562f7, only errors from the extension will be available with the button, which might be localized or not depending on the extension. The default implementation (for SQL) returns `undefined` and the button won't be visible. I think that should be ok now, since especially the [messageDetails](https://github.com/microsoft/azuredatastudio/blob/main/src/sql/platform/errorMessage/common/errorMessageService.ts#L19) documentation explicitely mentions technical information such as stack traces, so it shouldn't be unexpected if this information is not localized.